mmengine/dist/dist.py

```diff
@@ -308,7 +308,7 @@ def broadcast(data: Tensor,
         backend_device = get_comm_device(group)
         data_on_device = cast_data_device(data, backend_device)
```

Comment:
```suggestion
        data_on_device: Tensor = cast_data_device(data, backend_device)
```
This should make `mypy` happy